astropy/utils/introspection.py

```diff
@@ -136,30 +151,18 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
     else:
         raise ValueError('module argument must be an actual imported '
                          'module, or the import name of the module; '
-                         'got {!r}'.format(module))
-
-    if '.' not in version_path:
-        have_version = getattr(module, version_path)
-    else:
-        have_version = resolve_name(module.__name__, version_path)
-
-    # LooseVersion raises a TypeError when strings like dev, rc1 are part
-    # of the version number. Match the dotted numbers only. Regex taken
-    # from PEP440, https://www.python.org/dev/peps/pep-0440/, Appendix B
-    expr = '^([1-9]\\d*!)?(0|[1-9]\\d*)(\\.(0|[1-9]\\d*))*'
-    m = re.match(expr, version)
-    if m:
-        version = m.group(0)
+                         f'got {repr(module)}')
 
-    # have_version can have the same issue as version, so also regex it
-    m = re.match(expr, have_version)
-    if m:
-        have_version = m.group(0)
+    # Convert the module name to a distribution name
```

Comment:
I think we mostly do not need this dict, since for most packages the name and distribution name are the same. So, might it make sense to generally avoid creating it by just doing something like:
```
try:
    module_version  = metadata.version(module_name)
except metadata.PackageNotFoundError:  # module name different from package name?
    module_version = metadata.version(packages_distributions()[module_name][0])
```

Reply:
Good idea :+1: